src/Apollo/NcrReactionsProjector.php

```diff
@@ -0,0 +1,196 @@
+            $updateExpression .= " reactions.#{$reaction} :v_incr,";
+            $expressionAttributeNames["#{$reaction}"] = $reaction;
+        }
+
+        $params = [
+            'TableName'                 => $tableName,
+            'Key'                       => [
+                NodeTable::HASH_KEY_NAME => ['S' => $reactionsRef->toString()],
+            ],
+            'UpdateExpression'          =>  rtrim('add'. $updateExpression, ', '),
+            'ExpressionAttributeNames'  => $expressionAttributeNames,
+            'ExpressionAttributeValues' => [
+                ':v_incr' => ['N' => '1'],
+            ],
+            'ReturnValues'              => 'NONE',
+        ];
+
+        $this->client->updateItem($params);
+
+        // this ensures the ncr cache is current
+        // note that we don't put a new node as that would
+        // overwrite the atomic counting above.
+        $this->ncr->getNode($reactionsRef, true, $context);
+    }
+
+    protected function projectNode(Message $reactions, Message $event, Pbjx $pbjx): void
+    {
+        $context = ['causator' => $event];
+        $reactions
+            ->set('updated_at', $event->get('occurred_at'))
+            ->set('updater_ref', $event->get('ctx_user_ref'))
+            ->set('last_event_ref', $event->generateMessageRef())
+            ->set('etag', Aggregate::generateEtag($reactions));
+
+        $this->ncr->putNode($reactions, null, $context);
+
+        // Add empty reactions map to node
+        $reactionsRef = NodeRef::fromNode($reactions);
+        $tableName = $this->tableManager->getNodeTableName($reactionsRef->getQName(), $context);
+        $params = [
+            'TableName'                 => $tableName,
+            'Key'                       => [
+                NodeTable::HASH_KEY_NAME => ['S' => $reactionsRef->toString()],
+            ],
+            'UpdateExpression'          =>  'set reactions = :reactions_map',
+            'ExpressionAttributeValues' => [
+                ':reactions_map' => ['M' => []],
+            ],
+            'ReturnValues'              => 'NONE',
+        ];
+
+        $this->client->updateItem($params);
+        $pbjx->trigger($reactions, 'projected', new NodeProjectedEvent($reactions, $event), false, false);
+    }
+
+    protected function createReactions(NodeRef $nodeRef): Message
+    {
+        static $class = null;
+        if (null === $class) {
+            $class = MessageResolver::resolveCurie('*:apollo:node:reactions:v1');
+        }
+
+        $reactions = $class::fromArray(['_id' => $nodeRef->getId()]);
+        return $reactions;
+    }
+
+    protected function getReactions(NodeRef $nodeRef, Message $event): ?Message
+    {
+        try {
+            $reactionsRef = $this->createReactionsRef($nodeRef);
+            return $this->ncr->getNode($reactionsRef, true, ['causator' => $event]);
+        } catch (NodeNotFound $nf) {
+            // Node not found do nothing here
```

Comment:
@bhsiao55 nit: i'd probably just return null here, reads easier

Reply:
@bhsiao55 @efjacobson do you need the node?  hasNode is better and prolly doesn't need it's own method.  OR create the node if you get a record not found exception on update.

Reply:
@gdbrown I don't need the node, I will use hasNode instead